pkgs/top-level/python-packages.nix

```diff
@@ -6114,13 +6114,13 @@ self: super: with self; {
     IOKit = pkgs.darwin.apple_sdk_11_0.IOKit;
   };
 
-  jaxlib = self.jaxlib-build;
+  jaxlib = if self.jaxlib-build.meta.broken then self.jaxlib-bin else self.jaxlib-build;
```

Comment:
Ohhh I'm not sure how good of an idea this is: I can see myself tearing my hair out a few steps down the line (steps involving changing meta.unfree) trying to figure out how I ended up with an unfree pypi wheel in the closure instead of a cpu-only source-build

Reply:
I personally don't mind, but I get your point. Feel free to decide.

Reply:
I agree that this feels like a footgun. 

Another option: we alias `jaxlib = jaxlib-bin` and leave a comment linking to this PR and the hash mismatch issue.

Reply:
Ok, I like samuela's simple approach. Fixed.

Reply:
I'd suggest a more radical approach, which would probably cause frustration to other maintainers: we add a `lib.warn` to the `if broken then jaxlib-bin` branch saying that we're using a prebuilt and patchelfed jaxlib from pypi, potentially including unfree components

Reply:
It'd be annoying because it'll show warnings in the default nixpkgs instantiation and in ofborg

Reply:
And to be clear, the only admissible reason I see to add the alias is that this will get hydra and ofborg build reverse dependencies. Otherwise people could do this in a local overlay and we probably should do a better job advertising how to do overlays

Reply:
> The "if broken then jaxlib-bin else jaxlib-build" logic existed in the jax package.

Ohhh, I had completely missed that

Reply:
Ok, there was already an alias in place, and in this PR it place two roles:

- Supporting darwin
- Working around the hash issue

For the former, I'd replace the logic with `if jaxlib.meta.unsupported then jaxlib-bin else jaxlib` to look at `platforms` instead of `broken`. For the latter, I hope we can work around the instability soon.

> And if the build version is broken, the packages should keep working.

Yes but whether shipping a substitute "product" counts as "it's still working" is a matter of setting expectations.

Anyway I see now that my comment was out of scope

Reply:
> The "if broken then jaxlib-bin else jaxlib-build" logic existed in the jax package.

Yes, I wrote this in order to fix the build on darwin.

> Ok, there was already an alias in place

Actually, there is/was currently no such alias in place. The "if broken then jaxlib-bin else jaxlib-build" was only logic local to the jax package so that it would use jaxlib in `checkPhase`, but its impact was limited to that. See https://github.com/NixOS/nixpkgs/blob/6a8b9742401bba8771e5f8afe7b240b4410869cd/pkgs/development/python-modules/jax/default.nix#L29

Reply:
I think just writting ``if stdenv.isDarwin then A else B`` is way easier to understand and we should probably just go with that.